File: QuasarCode_Python/QuasarCode/Games/Dice/_NDice.py
from QuasarCode.Games.Dice import IDice
import random

class NDice(IDice):
    """
    A dice with a specified number of sides

    __init__():
        int sides --> number of sides on the dice
        int seed --> seed for the random generator (deafult is None)
    """
    def __init__(self, sides: int, seed: int = None):
        self.__Generator: random.Random = random.Random() if seed == None else random.Random(seed)

        # Number of sides on the dice
        self.Sides: int

    def Roll(self):
        """
        Rolls the dice
        """
        return self.__Generator.randint(1, self.Sides + 1)

    def MultipleDice(sides: list, noOfDice: int = len(sides)):
        """
        Convenience method for creating multiple dice

        Paramiters:
            list sides --> list of side numbers representing unique dice. For multiple of the same number, use a single element and alter noOfDice
            int noOfDice --> number of dice to create (deafult is length of sides list)
        """
        if len(sides) > 1:
            randomiser = random.Random()
            return [NDice(size, randomiser.Next(100000, 999999)) for size in sides]

        else:
            randomiser = random.Random()
            return [NDice(sides[0], randomiser.Next(100000, 999999)) for i in noOfDice]

File: QuasarCode_Library_Games/Dice/NDice.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace QuasarCode.Library.Games.Dice
{
    /// <summary>
    /// A dice with a specified number of sides
    /// </summary>
    public class NDice : IDice
    {
        private Random Generator;

        /// <summary>
        /// Number of sides on the dice
        /// </summary>
        public int Sides { get; }

        /// <summary>
        /// Creates a new NDice instance
        /// </summary>
        /// <param name="sides">Number of sides on the dice</param>
        public NDice(int sides)
        {
            Sides = sides;

            Generator = new Random();
        }

        /// <summary>
        /// Creates a new NDice instance
        /// </summary>
        /// <param name="sides">Number of sides on the dice</param>
        /// <param name="seed">Seed for the random generator</param>
        public NDice(int sides, int seed)
        {
            Sides = sides;

            Generator = new Random(seed);
        }

        /// <summary>
        /// Rolls the dice
        /// </summary>
        /// <returns>Integer outcome of the roll</returns>
        public int Roll()
        {
            return Generator.Next(1, Sides + 1);
        }


        /// <summary>
        /// Convenience method for creating multiple dice
        /// </summary>
        /// <param name="sides">The number of sides on each dice</param>
        /// <param name="noOfDice">Number of dice to create</param>
        /// <returns>Yeilds dice to produce an IEnumerable</returns>
        public static System.Collections.IEnumerable MultipleDice(int sides, int noOfDice)
        {
            Random randomiser = new Random();

            for (int i = 0; i < noOfDice; i++)
            {
                yield return new NDice(sides, randomiser.Next(100000, 999999));
            }
        }

        /// <summary>
        /// Convenience method for creating multiple dice
        /// </summary>
        /// <param name="sides">Side numbers representing each unique dice</param>
        /// <returns>Yeilds dice to produce an IEnumerable</returns>
        public static System.Collections.IEnumerable MultipleDice(params int[] sides)
        {
            Random randomiser = new Random();

            foreach (int size in sides)
            {
                yield return new NDice(size, randomiser.Next(100000, 999999));
            }
        }

        /// <summary>
        /// Convenience method for creating multiple dice
        /// </summary>
        /// <param name="sides">Collection of side numbers representing unique dice</param>
        /// <returns>Yeilds dice to produce an IEnumerable</returns>
        public static System.Collections.IEnumerable MultipleDice(ICollection<int> sides)
        {
            Random randomiser = new Random();

            foreach (int size in sides)
            {
                yield return new NDice(size, randomiser.Next(100000, 999999));
            }
        }
    }
}